From a942cccee6ff8e5d230f153d58f2345f0ec67289 Mon Sep 17 00:00:00 2001
From: Adam Stolcenburg <adam_stolcenburg@comcast.com>
Date: Thu, 12 Dec 2024 15:06:54 +0100
Subject: [PATCH] RIALTO-725 : Add the ability to define the Rialto socket
 location

Reason for change: Support for multiple Cobalt instances
using different Rialto sockets.
Test Procedure: See ticket
Risks: Low

Signed-off-by: Adam Stolcenburg <adam_stolcenburg@comcast.com>

Change-Id: I4a68c064920f479f508063ed90fd84cbb92ca2d7
---
 plugin/CobaltImplementation.cpp | 11 +++++++++++
 plugin/CobaltPlugin.json        |  4 ++++
 plugin/doc/CobaltPlugin.md      |  1 +
 3 files changed, 16 insertions(+)

diff --git a/plugin/CobaltImplementation.cpp b/plugin/CobaltImplementation.cpp
index f0036709..7dfc869d 100644
--- a/plugin/CobaltImplementation.cpp
+++ b/plugin/CobaltImplementation.cpp
@@ -114,6 +114,7 @@ private:
       Add(_T("advertisingid"), &AdvertisingId);
       Add(_T("closurepolicy"), &ClosurePolicy);
       Add(_T("fireboltendpoint"), &FireboltEndpoint);
+      Add(_T("rialtosocketname"), &RialtoSocketName);
       Add(_T("sbmainargs"), &SbMainArgs);
     }
     ~Config() {
@@ -132,6 +133,7 @@ private:
     Core::JSON::VariantContainer AdvertisingId;
     Core::JSON::String ClosurePolicy;
     Core::JSON::String FireboltEndpoint;
+    Core::JSON::String RialtoSocketName;
     Core::JSON::ArrayType<Core::JSON::String> SbMainArgs;
   };
 
@@ -339,6 +341,17 @@ private:
       if (config.FireboltEndpoint.IsSet() == true)
         Core::SystemInfo::SetEnvironment(_T("FIREBOLT_ENDPOINT"), config.FireboltEndpoint.Value());
 
+      if (config.RialtoSocketName.IsSet() == true) {
+        std::string dir;
+        if (Core::SystemInfo::GetEnvironment(_T("XDG_RUNTIME_DIR"), dir)) {
+          Core::SystemInfo::SetEnvironment(_T("RIALTO_SOCKET_PATH"), dir + "/" + config.RialtoSocketName.Value());
+          Core::SystemInfo::SetEnvironment(_T("GST_PLUGIN_SYSTEM_PATH"), _T("/tmp/gst"));
+          Core::SystemInfo::SetEnvironment(_T("GST_REGISTRY"), _T("/tmp/gst/registry.bin"));
+        } else {
+          SYSLOG(Logging::Notification, (_T("Ignore 'rialtosocketname' configuration, XDG_RUNTIME_DIR is not defined\n")));
+        }
+     }
+
       SYSLOG(Logging::Notification, (_T("Preload is set to: %s\n"), _preloadEnabled ? "true" : "false"));
 
       if (config.ClosurePolicy.IsSet() == true) {
diff --git a/plugin/CobaltPlugin.json b/plugin/CobaltPlugin.json
index 1a42cec6..e2b79145 100644
--- a/plugin/CobaltPlugin.json
+++ b/plugin/CobaltPlugin.json
@@ -107,6 +107,10 @@
             "type": "string",
             "description": "A URL that specifies access point to Firebolt Riple. Should include session id in the query."
           },
+          "rialtosocketname": {
+            "type": "string",
+            "description": "The name of the Unix socket inside the XDG_RUNTIME_DIR that represents the Rialto session."
+          },
           "advertisingid": {
             "$ref": "#/definitions/advertisingid"
           },
diff --git a/plugin/doc/CobaltPlugin.md b/plugin/doc/CobaltPlugin.md
index e76830ba..adb4fa25 100644
--- a/plugin/doc/CobaltPlugin.md
+++ b/plugin/doc/CobaltPlugin.md
@@ -94,6 +94,7 @@ The table below lists configuration options of the plugin.
 | configuration?.systemproperties?.friendlyname | string | <sup>*(optional)*</sup> A friendly name for this actual device |
 | configuration?.systemproperties?.devicetype | string | <sup>*(optional)*</sup> The type of the device. (must be one of the following: *SetTopBox*, *OverTheTopBox*, *TV*) |
 | configuration?.fireboltendpoint | string | <sup>*(optional)*</sup> A URL that specifies access point to Firebolt Riple. Should include session id in the query |
+| configuration?.rialtosocketname | string | <sup>*(optional)*</sup> The name of the Unix socket inside the XDG_RUNTIME_DIR that represents the Rialto session |
 | configuration?.advertisingid | object | <sup>*(optional)*</sup> Configure Identifier For Advertising |
 | configuration?.advertisingid?.ifa | string | <sup>*(optional)*</sup> Advertising ID or IFA |
 | configuration?.advertisingid?.lmt | string | <sup>*(optional)*</sup> Limit advertising tracking, treated as boolean |
-- 
2.43.0

